File: Java programming/problems on recursion/problem8.java
/*
8. Write a recursive program which accept number from user and return its factorial.  
	
	Input :  5 
	
	Output : 120 
	
	Author :Milind Nalawade
	
	Date : 25/08/2021
	
	Language : Java 14.0.2

*/

import java.lang.*;
import java.util.*;

class Logic
{
	public static int factor=1;
	static void Factor(int no)
	{
		if(no==1)
		{
			return ;
		}
		else
		{
			factor*=no;
			Factor(no-1);
		}
	}	
}

class problem8 extends Logic
{
	public static	void main(String argv[])
	{
		Scanner sobj=new Scanner(System.in);
		System.out.print("\n Input : ");
		int no=sobj.nextInt();
		Factor(no);
		System.out.print("\n Output : "+factor+"\n");
	}
}

File: C++ Programming/problems on recursion/problem8.cpp
/*
8. Write a recursive program which accept number from user and return its factorial.  
	
	Input :  5 
	
	Output : 120 
	
	Author :Milind Nalawade
	
	Date : 25/08/2021
	
	Language : C++ programming
	
*/
using namespace std;
#include<iostream>


//Logic
int factor=1;
int Factor(int no)
{
	if(no==1)
	{
		return factor;
	}
	else
	{
		factor*=no;
		no--;
		Factor(no);
	}
}	

//Entry point function
int main()
{
	int no=0;
	cout<<"\n Input : ";
	cin>>no;
	int ret=Factor(no);
	cout<<"\n Output : "<<ret<<"\n";
	return 0;
}	